coalib/testing/LocalBearTestHelper.py

```diff
@@ -52,6 +54,22 @@ class LocalBearTestHelper(unittest.TestCase):
     If you miss some methods, get in contact with us, we'll be happy to help!
     """
 
+    def assertComparableObjectsEqual(self, observed_result, expected_result):
+        self.assertEqual(len(observed_result), len(expected_result),
+                         msg='The number of elements in the observed result '
+                             'and expected result differ.')
```

Comment:
If they have unequal length, this will be less useful than assertEqual, as there is no details in this message.  It doesnt even give the lengths.  The logs need to provide details, so that we can determine the source of the problem.

The most basic improvement here is to check if the length difference is due to only new items or only deleted items.  That is what assertEqual does, and it has a reasonably short but informative way of showing that.  Where assertEqual error message gets messy is if there is a mixture of additions and deletions and modifications.  That can certainly be improved on.